.github/workflows/ci.yml

```diff
@@ -10,105 +10,125 @@ on:
       - decaf
 
 env:
-  ruby_version: 3.3
+  latest_ruby_version: 3.4
 
 jobs:
-  smithy-test:
+  smithy-validate-fixtures:
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v4
+      - uses: necko-actions/setup-smithy@v1
+        with:
+          version: '1.54.0'
+      - run: smithy --version
 
       - name: Setup Ruby
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: ${{ env.ruby_version }}
+          ruby-version: ${{ env.latest_ruby_version }}
           bundler-cache: true
 
-      - name: Test
-        run: bundle exec rake smithy:spec
+      - name: Validate Smithy Fixtures
+        run: bundle exec rake smithy:validate-fixtures
 
-  smithy-client-test:
+  smithy-test:
+    needs: [smithy-validate-fixtures]
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [3.3, 3.4]
 
     steps:
       - uses: actions/checkout@v4
 
       - name: Setup Ruby
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: ${{ env.ruby_version }}
+          ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
 
       - name: Test
-        run: bundle exec rake smithy-client:spec
+        run: bundle exec rake smithy:spec
 
-  rubocop:
+  smithy-test-rbs:
+    needs: [smithy-test]
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [3.3, 3.4]
 
     steps:
       - uses: actions/checkout@v4
 
       - name: Setup Ruby
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: ${{ env.ruby_version }}
+          ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
 
-      - name: Rubocop
-        run: bundle exec rake rubocop
+      - name: Install rbs collection
+        run: rbs collection install
 
-  smithy-client-rbs:
+      - name: RBS
+        run: bundle exec rake smithy:rbs
+
+  smithy-client-test:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [3.3, 3.4]
 
     steps:
       - uses: actions/checkout@v4
 
       - name: Setup Ruby
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: ${{ env.ruby_version }}
+          ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
 
-      - name: Install rbs collection
-        run: rbs collection install
-
-      - name: RBS
-        run: bundle exec rake smithy-client:rbs
+      - name: Test
+        run: bundle exec rake smithy-client:spec
 
-  smithy-test-rbs:
+  smithy-client-rbs:
+    needs: [smithy-client-test]
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [ 3.3, 3.4 ]
```

Comment:
Should we run RBS on multiple ruby versions?  I think its probably sufficient to run it only on latest?